34.md

```diff
@@ -0,0 +1,84 @@
+NIP-34
+======
+
+`git` stuff
+-----------
+
+`draft` `optional`
+
+This NIP defines all the ways code collaboration using and adjacent to [`git`](https://git-scm.com/) can be done using Nostr.
+
+## Repository announcements
+
+Git repositories are hosted in Git-enabled servers, but their existence can be announced using Nostr events, as well as their willingness to receive patches, bug reports and comments in general.
+
+```jsonc
+{
+  "kind": 30617,
+  "content": "",
+  "tags": [
+    ["d", "<repo-id>"],
+    ["name", "<human-readable project name>"],
+    ["description", "brief human-readable project description>"],
+    ["head", "<sha1-hex>"], // the current repository commit head, as hex
+    ["web", "<url for browsing>"], // a webpage url, if the git server being used provides such a thing
+    ["clone", "<url for git-cloning>"], // a url to be given to `git clone` so anyone can clone it
+    ["patches", "<relay-url>"], // this signals that patches are welcome and will be searched for in this relay
+    ["issues", "<relay-url>"] // this signals that bug reports, questions, comments and so on are welcome
+  ]
+}
+```
+
+The tags `web`, `clone`, `patches` and `issues` can be specified multiple times. Except `d`, all tags are optional.
+
+## Patches
+
+Patches can be sent by anyone to any repository. Patches to a specific repository SHOULD be sent to the relays specified in that repository's announcement event's `"patches"` tag. Patch events SHOULD include an `a` tag pointing to that repository's announcement address.
+
+```jsonc
+{
+  "kind": 1617,
+  "content": "<patch>", // contents of <git format-patch>
+  "tags": [
+    ["a", "30617:<base-repo-owner-pubkey>:<base-repo-id>"],
+    ["p", "<repository-owner>"]
+  ]
+}
+```
+
+## Issues
+
+Issues are Markdown text that is just human-readable conversational threads related to the repository: bug reports, feature requests, questions or comments of any kind.
+
+```jsonc
+{
+  "kind": 1621,
+  "content": "<markdown text>",
+  "tags": [
+    ["a", "30617:<base-repo-owner-pubkey>:<base-repo-id>"],
+    ["p", "<repository-owner>"]
+  ]
+}
+```
+
+## Replies
+
+Replies are also Markdown text. The difference is that they MUST refer to one root thread event. Their root doesn't have to be a `kind:1621` _issue_, but also a `kind:1617` patch event.
```

Comment:
I think the MUST restriction on the root thread is a bit much, you should be able to reply to patch 02/08 in a nip10 thread of patches no?

Reply:
unless this means at least one root thread event along with the thing you're replying to. maybe this could be mentioned here.

Reply:
the root item could be the cover letter. see `git format-patch --cover-letter ..branch` for an example.

Reply:
Changed the text of this. See if it sounds better now.